### PR TITLE
Opt/rot acorr

### PR DESCRIPTION
**Pull request by vyasr**
Further optimizations for rotational autocorrelation. Gives another 5x speedup over the old method. Also gets rid of the usage of signed ints for quantities that should be unsigned.

### CODE REVIEW

File: cpp/order/RotationalAutocorrelation.h [hunk 1]
```diff
@@ -7,7 +7,9 @@
 #include <cassert>
 #include <complex>
 #include <memory>
+#include <iostream>
```
Comment (bdice) [inline cpp/order/RotationalAutocorrelation.h]: Are `<iostream>` and `<map>` still needed? Remove if unused. Then this is good to go.

**Review by bdice — COMMENTED** [review]
Looks good to me, one edit.

File: cpp/order/RotationalAutocorrelation.h (hunk 1 above)
Comment (vyasr) [reply cpp/order/RotationalAutocorrelation.h]: Nope, I noticed those and then left them for you to find as a test :)

Comment (bdice) [reply cpp/order/RotationalAutocorrelation.h]: @vyasr I passed! 🥇

**Review by bdice — APPROVED** [review]
Looks good to me!

**Pull request merged**